python/psautohint/__main__.py

```diff
@@ -717,5 +734,180 @@ def main(args=None):
+             'The file must contain a comma-separated sequence of glyph '
+             'identifiers.'
+    )
+    glyphs_parser.add_argument(
+        '-x',
+        '--exclude-glyphs',
+        metavar='GLYPH_LIST',
+        dest='glyphs_to_not_hint',
+        type=_split_comma_sequence,
+        default=[],
+        help='comma-separated sequence of glyphs to NOT process\n'
+             "Counterpart to '--glyphs' option."
+    )
+    glyphs_parser.add_argument(
+        '--exclude-glyphs-file',
+        metavar='PATH',
+        dest='glyphs_to_not_hint_file',
+        type=_validate_path,
+        help='file containing a list of glyphs to NOT process\n'
+             "Counterpart to '--glyphs-file' option."
+    )
+    parser.add_argument(
+        '-v',
+        '--verbose',
+        action='count',
+        default=0,
+        help='verbose mode\n'
+             'Use -vv for extra-verbose mode.'
+    )
+    parser.add_argument(
+        '--version',
+        action='version',
+        version=__version__
+    )
+    parser.add_argument(
+        '--traceback',
+        action='store_true',
+        help="show traceback for exceptions.",
+    )
+    parsed_args = parser.parse_args(args)
+
+    if parsed_args.verbose == 0:
+        log_level = logging.WARNING
+    elif parsed_args.verbose == 1:
+        log_level = logging.INFO
+    else:
+        log_level = logging.DEBUG
+
+    logging.basicConfig(format="%(levelname)s: %(message)s", level=log_level)
+
+    if (parsed_args.output_paths and
+            len(parsed_args.font_paths) != len(parsed_args.output_paths)):
+        parser.error("number of input and output fonts differ")
+
+    if len(parsed_args.font_paths) != len(set(parsed_args.font_paths)):
+        parser.error("found duplicate font inputs")
+
+    options = ReportOptions(parsed_args)
+
+    options.font_format = _validate_font_paths(parsed_args.font_paths, parser)
+
+    if parsed_args.glyphs_to_hint:
+        options.glyphList = _process_glyph_list_arg(
+            parsed_args.glyphs_to_hint, options.nameAliases)
+    elif parsed_args.glyphs_to_not_hint:
+        options.excludeGlyphList = True
+        options.glyphList = _process_glyph_list_arg(
+            parsed_args.glyphs_to_not_hint, options.nameAliases)
+    elif parsed_args.glyphs_to_hint_file:
+        options.glyphList = _process_glyph_list_arg(
+            _read_txt_file(parsed_args.glyphs_to_hint_file),
+            options.nameAliases)
+    elif parsed_args.glyphs_to_not_hint_file:
+        options.excludeGlyphList = True
+        options.glyphList = _process_glyph_list_arg(
+            _read_txt_file(parsed_args.glyphs_to_not_hint_file),
+            options.nameAliases)
+
+    return options, parsed_args
+
+
+def stemhist(args=None):
+    options, pargs = get_stemhist_options(args)
+
+    try:
+        hintFiles(options)
+    except Exception as ex:
+        if pargs.traceback:
+            raise
+        logging.error(ex)
+        sys.exit(1)
```

Comment:
it's better that you don't raise SystemExit inside here or similar main-like functions that are used as console scripts entry points. Just return the exit code `return 1`, and the wrapper will raise `sys.exit(your_entry_point())` for you. Otherwise when you call stemhist in the test suite, and it fails, it will stop the whole test run and pytest process will exit abruptly (see `test_cli.py::test_stemhist` where you call `stemhist` function without catching SystemExit)

Reply:
Aha, that is why the logs were truncated in the failing builds. Thanks!